src/pybamm/models/full_battery_models/sodium_ion/basic_dfn.py

```diff
@@ -0,0 +1,273 @@
+#
+# Basic Doyle-Fuller-Newman (DFN) Model
+#
+import pybamm
+
+
+class BasicDFN(pybamm.lithium_ion.BaseModel):
+    """Doyle-Fuller-Newman (DFN) model of a sodium-ion battery, from
+    :footcite:t:`Marquis2019`.
+
+    Parameters
+    ----------
+    name : str, optional
+        The name of the model.
+
+    """
+
+    def __init__(self, name="Doyle-Fuller-Newman model"):
+        super().__init__(name=name)
+        pybamm.citations.register("Marquis2019")
+        # `param` is a class containing all the relevant parameters and functions for
+        # this model. These are purely symbolic at this stage, and will be set by the
+        # `ParameterValues` class when the model is processed.
+        param = self.param
```

Comment:
if these are the same, then why not just use self.param instead?

Reply:
Actually, I just copied this from the lithium_ion BasicDFN. I believe we do that in all the basic models, so we don't have to do self.param every time (there any a lot of occurrences) and the code is easier to read. I personally think it is useful as it is easier to read through the model.

Reply:
There's 63 occurrences of `param = self.param` across models, so will open a separate issue & PR for that (will also make it easier to review).